.github/workflows/lint-pull-request.yml

```diff
@@ -0,0 +1,17 @@
+name: 'Lint PR'
+
+on:
+  pull_request_target:
```

Comment:
Not sure if there's a way to scope this down to just v4, but with push targets you can do something like this to specify branches: https://github.com/PyreStudios/steward/blob/main/.github/workflows/deploy_docs.yml#L5